File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    for i in range(1, number+1):
        count = 0
        for k in range(1, int(i**(1/2)) + 1):
            
            if i % k == 0:
                count+=1
                if ((k**2) != i):
                    count+=1
        if count > limit:
            count = power
        answer += count
    return answer

File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    const answer = [];
    const divisor = (num) => {
        const div_arr = [];
        for(let i=1; i < parseInt(num**(1/2)) + 1; i++){
            if(num % i === 0){
                div_arr.push(i);
                if(i**2 !== num){
                    div_arr.push(parseInt(num / i));
                };
            };
        };
        return div_arr.length;
    };
    for(let i=1; i <= number; i++){
        let div_count = divisor(i);
        answer.push((div_count > limit) ? power : div_count);
    };
    return answer.reduce((a,b) => a+b);
}